compiler/dyno/include/chpl/queries/session-queries.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```

Comment:
Might be nice to have some doc guidance on what kind of queries we expect here, or the context that explains the meaning of a "session".

Reply:
Other terms might be "compiler invocation" or maybe "compiler configuration". I don't have any strong preference on the name right now but I think I slightly prefer something more along the lines of "compilation configuration". We have some other stuff in parsing-queries.h that arguably would belong in a "compiler configuration" area -- setModuleSearchPath / setBundledModulePath and related things.

Reply:
Ok, is it fine for this header to live in `chpl/queries`? Do you want `-queries.h` at the end of it to follow convention? Or just `compiler-configuration.h`?

Reply:
I have been thinking maybe it's not a good practice to have `bla-queries.h` since it's just a pile of functionality and whether it is a query or not is up to individual functions. I'm OK with putting it in `chpl/queries` but arguably it is off-topic there. OK for now though. Maybe `chpl/compiler` (including a new namespace) or something would be called for.